internal/dispatch/graph/computecheck_test.go

```diff
@@ -0,0 +1,506 @@
+				{core.RelationTupleUpdate_CREATE, "document:foo#org@organization:someorg", "anothercaveat", nil},
+				{core.RelationTupleUpdate_CREATE, "document:bar#org@organization:someorg", "", nil},
+				{core.RelationTupleUpdate_CREATE, "document:foo#editor@user:vic", "testcaveat", map[string]any{"somecondition": 42, "somebool": true}},
+				{core.RelationTupleUpdate_CREATE, "document:foo#viewer@user:vic", "testcaveat", map[string]any{"somecondition": 42, "somebool": true}},
+				{core.RelationTupleUpdate_CREATE, "document:foo#viewer@user:blippy", "testcaveat", map[string]any{"somecondition": 42, "somebool": "hi"}},
+				{core.RelationTupleUpdate_CREATE, "document:foo#viewer@user:noa", "testcaveat", map[string]any{"somecondition": 42, "somebool": "hi"}},
+				{core.RelationTupleUpdate_CREATE, "document:foo#editor@user:noa", "testcaveat", map[string]any{"somecondition": 42, "somebool": "hi"}},
+				{core.RelationTupleUpdate_CREATE, "document:foo#editor@user:wayne", "invalid", nil},
+			},
+			[]check{
+				{
+					"document:foo#view@user:sarah",
+					nil,
+					v1.ResourceCheckResult_CAVEATED_MEMBER,
+					[]string{"anothercondition"},
+					"",
+				},
+				{
+					"document:foo#view@user:sarah",
+					map[string]any{
+						"somecondition": 42,
+					},
+					v1.ResourceCheckResult_CAVEATED_MEMBER,
+					[]string{"anothercondition"},
+					"",
+				},
+				{
+					"document:foo#view@user:sarah",
+					map[string]any{
+						"anothercondition": 15,
+					},
+					v1.ResourceCheckResult_CAVEATED_MEMBER,
+					[]string{"somecondition"},
+					"",
+				},
+				{
+					"document:foo#view@user:sarah",
+					map[string]any{
+						"somecondition":    42,
+						"anothercondition": 15,
+						"somebool":         true,
+					},
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:foo#view@user:john",
+					map[string]any{
+						"anothercondition": 14,
+					},
+					v1.ResourceCheckResult_NOT_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:foo#view@user:john",
+					map[string]any{
+						"anothercondition": 15,
+					},
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:bar#view@user:jane", nil, v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:foo#view@user:peter",
+					nil,
+					v1.ResourceCheckResult_NOT_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:foo#view@user:vic",
+					nil,
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"",
+				},
+				{
+					"document:foo#view@user:blippy",
+					nil,
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"no such overload",
+				},
+				{
+					"document:foo#view@user:noa",
+					nil,
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"no such overload",
+				},
+				{
+					"document:foo#view@user:wayne",
+					nil,
+					v1.ResourceCheckResult_MEMBER,
+					nil,
+					"caveat with name `invalid` not found",
+				},
```

Comment:
- test error during caveat evaluation without children
- test error during caveat evaluation with children
- test invalid caveat. this is possible because we are bypassing the checks in the service controller